src/hotspot/cpu/riscv/macroAssembler_riscv.hpp

```diff
@@ -60,9 +60,12 @@ class MacroAssembler: public Assembler {
   }
 
   void leave() {
-    mv(sp, fp);
-    ld(fp, Address(sp, -2 * wordSize));
-    ld(ra, Address(sp, -wordSize));
+    // Note: setting sp above the address of alive variables could result in undefined behaviors:
```

Comment:
I would suggest move this code comment before function "void enter()" in the same file.
Maybe rephrasing like this:
"SP must be updated to the right place before saving/restoring RA and FP because signal
 based thread suspend/resume could happen asynchronously."

Reply:
Thanks for the suggestion, changed.

Reply:
I think it's better to go like this:

  // Stack frame creation/removal
  // Note that SP must be updated to the right place before saving/restoring RA and FP
  // because signal based thread suspend/resume could happen asynchronously.

Reply:
Thank you, changed.